test/integration/client-side-encryption/driver.test.ts

```diff
@@ -471,3 +476,89 @@ describe('Range Explicit Encryption with JS native types', function () {
     });
   });
 });
+
+describe('CSOT', function () {
+  describe('Explicit Encryption', function () {
+    describe('#createEncryptedCollection', function () {
+      let keyVaultClient: MongoClient;
+      let internalClient: MongoClient;
+      let clientEncryption: ClientEncryption;
+      const LOCAL_KEY = Buffer.from(
+        'Mng0NCt4ZHVUYUJCa1kxNkVyNUR1QURhZ2h2UzR2d2RrZzh0cFBwM3R6NmdWMDFBMUN3YkQ5aXRRMkhGRGdQV09wOGVNYUMxT2k3NjZKelhaQmRCZGJkTXVyZG9uSjFk',
+        'base64'
+      );
+
+      beforeEach(async function () {
+        internalClient = this.configuration.newClient();
+        await internalClient.connect();
+        await internalClient.db('keyvault').createCollection('datakeys');
+        keyVaultClient = this.configuration.newClient({ timeoutMS: 100 });
+        clientEncryption = new ClientEncryption(keyVaultClient, {
+          keyVaultNamespace: 'keyvault.datakeys',
+          kmsProviders: { local: { key: LOCAL_KEY } }
+        });
+        await internalClient
+          .db()
+          .admin()
+          .command({
+            configureFailPoint: 'failCommand',
+            mode: {
+              times: 1
+            },
+            data: {
+              failCommands: ['create'],
+              blockConnection: true,
+              blockTimeMS: 1000
+            }
+          } as FailPoint);
+      });
+
+      afterEach(async function () {
+        await internalClient
+          .db()
+          .admin()
+          .command({
+            configureFailPoint: 'failCommand',
+            mode: 'off'
+          } as FailPoint);
+        await internalClient
+          .db('db')
+          .collection('newnew')
+          .drop()
+          .catch(() => null);
+        await internalClient.close();
+        await keyVaultClient.close();
+      });
+
+      it(
+        'times out due to timeoutMS',
+        {
+          requires: {
+            clientSideEncryption: true,
+            mongodb: '>=7.0.0',
+            topology: '!single'
+          }
+        },
+        async function () {
+          const createCollectionOptions = {
```

Comment:
For composite operations (operations made up of multiple things that can time out), it is important that we test that

- timeoutMS applies to all operations cumulatively.  In this case, it needs to apply to all `createDataKey` invocations as well as in addition to the final createCollection
- timeoutMS applies to each async operation

We usually test this by

- making each async operation time out, confirming that each async op respects timeoutMS
- setting failcommands on all async operations such that in total they block for longer than `timeoutMS`.  In this case, that could look like a fail command on insert and createCollection, blocking for 500MS each, with timeoutMS of like 800ms.  Neither operation times out individually, but collectively they time out.

Finally, its important to measure the duration and confirm it fails in roughly the amount of time we expect (timeoutMS).